packages/@aws-cdk/aws-codepipeline-actions/README.md

```diff
@@ -764,6 +764,31 @@ const deployStage = pipeline.addStage({
 
 [image definition file]: https://docs.aws.amazon.com/codepipeline/latest/userguide/pipelines-create.html#pipelines-create-image-definitions
 
+#### Deploying ECS applications with existing ECS service ARN
+
+CodePipeline can deploy an ECS service which [cluster formatted ECS service ARN](https://docs.aws.amazon.com/AmazonECS/latest/developerguide/ecs-account-settings.html#ecs-resource-ids) this can be deploy across region and accounts as well using the ARN.
+
+```ts
+import * as ecs from `@aws-cdk/aws-ecs`;
+
+const service = ecs.BaseService.fromServiceArnWithCluster(this, 'EcsService',
+  'arn:aws:ecs:us-east-1:123456789012:service/myClusterName}/myServiceName'
+);
+const pipeline = new codepipeline.Pipeline(this, 'MyPipeline');
+const buildOutput = new codepipeline.Artifact();
+const deployStage = pipeline.addStage({
+  stageName: 'Deploy',
+  actions: [
+    new codepipeline_actions.EcsDeployAction({
+      actionName: 'DeployAction',
+      service: service,
+      input: buildOutput,
+    }),
```

Comment:
We might want to add an note in here that for cross accounts you will want to provide the role, because if not a role name will be generated for you which doesn't actually get created, what do you think?

Reply:
Hmm, not sure I understand what you mean...?

If the Action is cross-account, a new Role, with a well-known name, will be generated for you in the correct account (if you don't specify the `role` property).

Reply:
If you don't provide role on the ECSDeployAction a roleArn will be provided in the code pipeline actions for you.
If you are deploying across accounts, that role will most likely not exist in the other account.

From my ecs deploy action test.
```
                RoleArn: {
                  'Fn::Join': [
                    '',
                    [
                      'arn:',
                      {
                        Ref: 'AWS::Partition',
                      },
                      ':iam::service-account:role/pipelinestack-support-serloyecsactionrole49867f847238c85af7c0',
                    ],
                  ],
```

In order for the deployment to work I would need to create an IAM Role with the name `pipelinestack-support-serloyecsactionrole49867f847238c85af7c0` in the service account, in order for the pipeline to be able to deploy.

In a self-mutating CDK pipeline the self mutation will actually fail because the account doesn't exist so it can't update the Policy for the KMS key, bucket and Pipeline Role's Policy.  I would guess an normal pipeline might have the same issue as well.

Providing an IAM role which is already created in the other account resolves that, typically we have been leveraging the CDK bootstrap deploy role and adding permissions to deploy ECS matching [this](https://github.com/aws/aws-cdk/blob/d735ce4bc9049b68b830641d68c97f4be565968d/packages/%40aws-cdk/aws-codepipeline-actions/lib/ecs/deploy-action.ts#L85-L110) is typically how we have been working around this.  [Write up from other PR on role](https://github.com/aws/aws-cdk/pull/18059/files#diff-f6cb9a2b0512ad244db5d9f4349017a38d7cc096cddf1989a8047a7e45dd8635R52-R98).

Reply:
I mean, that's not just some random Role name that we just make up 🙂. There is another Stack in the application generated, in the target account, that contains that Role, and the Pipeline Stack depends on that Stack, so `cdk deploy PipelineStack` _should_ correctly include it, deploy it first, and then everything should work.

If it doesn't, that's a bug we need to fix.

However, it's fine if you want to mention the Role in the ReadMe.

Reply:
Interesting, I guess I never noticed that other support stack being created for the service account, to create that role.  I have see the other stack which is in the other region from the pipeline which gets generated in the pipeline account.  I added a little blurb about the role, I know I have seen issues with the CDK pipeline when adding a new region having an issue if the role referenced hasn't been created.